File: app/src/main/java/com/riders/thelab/data/remote/dto/weather/CurrentWeather.java
package com.riders.thelab.data.remote.dto.weather;

import com.squareup.moshi.Json;

import java.util.List;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Setter
@Getter
@ToString
public class CurrentWeather {
    @Json(name = "dt")
    private long dateTimeUTC;

    @Json(name = "sunrise")
    private long sunrise;

    @Json(name = "sunset")
    private long sunset;

    @Json(name = "moonrise")
    private long moonrise;

    @Json(name = "moonset")
    private long moonset;

    @Json(name = "moon_phase")
    private double moonPhase;

    @Json(name = "temp")
    private double temperature;

    @Json(name = "feels_like")
    private double feelsLike;

    @Json(name = "pressure")
    private int pressure;

    @Json(name = "humidity")
    private int humidity;

    @Json(name = "dew_point")
    private double dewPoint;

    // UV index
    @Json(name = "uvi")
    private double UVIndex;

    @Json(name = "clouds")
    private int clouds;

    // Average visibility, metres
    @Json(name = "visibility")
    private int visibility;

    @Json(name = "wind_speed")
    private double windSpeed;

    @Json(name = "wind_deg")
    private int windDegree;

    @Json(name = "wind_gust")
    private double windGust;

    @Json(name = "weather")
    private List<Weather> weather;

    @Json(name = "rain")
    private Rain rain;

    @Json(name = "snow")
    private Snow snow;

    // Probability of precipitation
    @Json(name = "pop")
    private double pop;
}

File: app/src/main/java/com/riders/thelab/data/remote/dto/weather/CurrentWeather.kt
package com.riders.thelab.data.remote.dto.weather

import kotlinx.serialization.SerialName
import java.io.Serializable

@kotlinx.serialization.Serializable
data class CurrentWeather(
    @SerialName("dt")
    val dateTimeUTC: Long = 0,

    @SerialName("sunrise")
    val sunrise: Long = 0,

    @SerialName("sunset")
    val sunset: Long = 0,

    @SerialName("moonrise")
    val moonrise: Long = 0,

    @SerialName("moonset")
    val moonset: Long = 0,

    @SerialName("moon_phase")
    val moonPhase: Double = 0.0,

    @SerialName("temp")
    val temperature: Double = 0.0,

    @SerialName("feels_like")
    val feelsLike: Double = 0.0,

    @SerialName("pressure")
    val pressure: Int = 0,

    @SerialName("humidity")
    val humidity: Int = 0,

    @SerialName("dew_point")
    val dewPoint: Double = 0.0,

    // UV index
    @SerialName("uvi")
    val uvIndex: Double = 0.0,

    @SerialName("clouds")
    val clouds: Int = 0,

    // Average visibility, metres
    @SerialName("visibility")
    val visibility: Int = 0,

    @SerialName("wind_speed")
    val windSpeed: Double = 0.0,

    @SerialName("wind_deg")
    val windDegree: Int = 0,

    @SerialName("wind_gust")
    val windGust: Double = 0.0,

    @SerialName("weather")
    val weather: List<Weather>? = null,

    @SerialName("rain")
    val rain: Rain? = null,

    @SerialName("snow")
    val snow: Snow? = null,

    // Probability of precipitation
    @SerialName("pop")
    val pop: Double = 0.0
) : Serializable {
    constructor() : this(
        0,
        0,
        0,
        0,
        0,
        0.0,
        0.0,
        0.0,
        0,
        0,
        0.0,
        0.0,
        0,
        0,
        0.0,
        0,
        0.0,
        null,
        null,
        null,
        0.0
    )
}
